### ISSUE TITLE
Support published ports

### ISSUE THREAD

**Issue by hjr3** [purton-tech/rust-on-nails#30]
First, thank you for start this project.

I think most VSCode users are using https://code.visualstudio.com/docs/devcontainers/containers#_temporarily-forwarding-a-port to forward port 3000 to their host machine. I am not using VSCode for my dev container setup though.

To get my setup working, I have to make the following changes:

Publish container port 3000 to host IP 127.0.0.1, port 3000.
```patch
--- a/.devcontainer/docker-compose.yml
+++ b/.devcontainer/docker-compose.yml
@@ -13,10 +13,12 @@ services:
       retries: 5

   development:
-    build:
+    build:
       context: .
       dockerfile: Dockerfile
-
+    ports:
+      - "127.0.0.1:3000:3000"
+
     volumes:
       - ..:/workspace:cached
       # Give access to ssh keys on the host (Is there a better way to do this?)
```

Make the server inside the container accept connections on all (virtual) interfaces rather than just the container's localhost (127.0.0.1).
```patch
--- a/crates/axum-server/src/main.rs
+++ b/crates/axum-server/src/main.rs
@@ -19,7 +19,7 @@ async fn main() {
         .layer(Extension(pool.clone()));

     // run it
-    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
+    let addr = SocketAddr::from(([0, 0, 0, 0], 3000));
     println!("listening on {}", addr);
     axum::Server::bind(&addr)
         .serve(app.into_make_service())
```

From a Microsoft/GitHub developer:

> Our examples show using 0.0.0.0 instead of localhost ...

[source](https://github.com/microsoft/vscode-remote-release/issues/1009#issuecomment-518883583)

This was in 2019 though and I know they have since separated the notion of _always forward a port_ and _publish a port_.

I absolutely understand if you want to leave things as is. If so, this can serve as guidance for anyone else who runs into this issue.

**Comment by ianpurton** [comment]
@hjr3 Can I ask how you are using the devcontainer ? Are you using docker-compose?

I used to publish ports, but then yes you're right vscode kind of handles that. But I could add a section on how to run with docker-compose, or if you let me know how you do it, I can add that.

There's also https://containers.dev/ which has a command line for running a devcontainer. I don't really want people to have to install too much on their local machine, but I could add a section on that too.

I'll make the change for 0.0.0.0

**Comment by hjr3** [comment]
Thank you for the CLI tip. This worked perfectly:

```
$ npx @devcontainers/cli up --workspace-folder=.
$ npx @devcontainers/cli exec --workspace-folder=. bash
```

I do not think we need to add a section for this unless a lot more people use this method.

I saw you landed e8630638cee4b5df60344ffecc4679dca2d34491 but there are a few other places we listen on localhost. I opened https://github.com/purton-tech/rust-on-nails/pull/45 to fix the rest.
